libr/core/cmd_anal.c

```diff
@@ -4419,16 +4417,14 @@ R_API int r_core_esil_step(RCore *core, ut64 until_addr, const char *until_expr,
 	RAnalOp op = {0};
 	RAnalEsil *esil = core->anal->esil;
 	const char *name = r_reg_get_name (core->anal->reg, R_REG_NAME_PC);
+	ut64 addr;
 	bool breakoninvalid = r_config_get_i (core->config, "esil.breakoninvalid");
 	int esiltimeout = r_config_get_i (core->config, "esil.timeout");
 	ut64 startTime;
-	if (!esil) {
```

Comment:
in case esil is null, that will be crashing, if you dont run  'aeim' before 'aes' for example

Reply:
There's another check down below to handle that situation
https://github.com/radareorg/radare2/blob/05ee0962800a8afa1b609e8d08ff87b3b986c357/libr/core/cmd_anal.c#L4447

Reply:
```
$> r2 /bin/ls
 -- Beer in mind.
[0x000067d0]> aes
[0x00006874]> 
```